### PR TITLE
Add World Clone example

### PR DESCRIPTION
**Pull request by caspark**
And expose necessary APIs.

Based on discussions in #372.

### CODE REVIEW

**Review by Ralith — APPROVED** [review]
LGTM, thanks! Want to update the changelog?

**Comment by caspark** [comment]
Done! Added a tiny bit more docs too.

File: src/lib.rs [hunk 1]
```diff
@@ -84,7 +84,7 @@ pub mod serialize;
 mod take;
 mod world;
 
-pub use archetype::{Archetype, ArchetypeColumn, ArchetypeColumnMut};
+pub use archetype::{Archetype, ArchetypeColumn, ArchetypeColumnMut, TypeIdMap, TypeInfo};
```
Comment (Ralith) [inline src/lib.rs:87]: This should also be noted in the changelog.

**Review by Ralith — APPROVED** [review]
Thanks!

**Pull request merged**